music21/metadata/__init__.py

```diff
@@ -308,48 +743,281 @@ def all(self, skipContributors=False):
+                    and isinstance(value, t.Iterable)
+                    and not isinstance(value, str)):
+                raise ValueError(f'md.{name} can only be set to a single value; '
+                                 f'set md[{name}] to multiple values instead.')
+            self._set(properties.MUSIC21_WORK_ID_TO_NSKEY[name], value, isCustom=False)
+            return
+
+        # Is name a grandfathered workId abbreviation?
+        if name in properties.MUSIC21_ABBREVIATION_TO_NSKEY:
+            if (value is not None
+                    and isinstance(value, t.Iterable)
+                    and not isinstance(value, str)):
+                raise ValueError(f'md.{name} can only be set to a single value; '
+                                 f'set md[{name}] to multiple values instead.')
+            self._set(properties.MUSIC21_ABBREVIATION_TO_NSKEY[name], value, isCustom=False)
+            return
+
+        # Is name one of the non-uniqueName property.setters (i.e the three
+        # plural contributors, and the three new fileInfo setters)?
+        if name in ('composers', 'librettists', 'lyricists'):
+            uniqueName = name[:-1]  # remove the trailing 's'
+            # check that value is t.Iterable (and not str)
+            if not isinstance(value, t.Iterable) or isinstance(value, str):
+                raise ValueError(f'md.{name} can only be set to multiple values.')
+            self._set(uniqueName, value, isCustom=False)
+            return
+
+        if name == 'fileFormat':
+            if value is None:
+                self.fileInfo.format = None
+            else:
+                self.fileInfo.format = Text(value)
+            return
+
+        if name == 'filePath':
+            if value is None:
+                self.fileInfo.path = None
+            else:
+                self.fileInfo.path = Text(value)
+            return
+
+        if name == 'fileNumber':
+            if value is None:
+                self.fileInfo.number = None
+            else:
+                self.fileInfo.number = int(value)
+            return
+
+        # OK, we've covered everything we know about; fall back to setting
+        # bare attributes (including the ones in base classes).
+        super().__setattr__(name, value)
+
+    def __getitem__(self, key: str) -> t.Tuple[ValueType, ...]:
+        '''
+        Utility "dictionary key" access for all standard uniqueNames and
+        standard keys of the form 'namespace:name'.
+
+        These always return t.Tuple[ValueType, ...], which may be empty.
+
+        If key is not a standard uniqueName or standard 'namespace:name',
+        then KeyError is raised.
+
+        >>> md = metadata.Metadata()
+        >>> md['description'] = [metadata.Text('A fun score!', language='en')]
+        >>> descs = md['description']
+        >>> descs
+        (<music21.metadata.primitives.Text A fun score!>,)
+        >>> md.add('description', 'Also a great tune')
```

Comment:
maybe a little bit more serious:

description 1: 'For the coronation of Catherine the Great.'
description 2: 'In five sections, unique for its time.'

to give people a sense of why they might want to have two different description objects.

I didn't join Twitter for years because of their example: "Use twitter to tell friends what you're up to: Example: 'I am eating soup'."  -- true story.  I just thought, "I don't need a platform to tell people that I'm eating soup."  Good examples go a long way to explaining the value of a function.

Reply:
Done.